tox.ini

```diff
@@ -89,7 +89,9 @@ commands = {posargs}
 basepython = python3
 deps = -r{toxinidir}/build-requirements.txt
 commands =
-    charmcraft build
+    charmcraft clean
+    charmcraft -v pack
+    charmcraft clean
```

Comment:
You're calling `charmcraft clean` twice. This is wrong - here's where you typically use a shell script to rename the resulting charm so that you can trim all the unnecessary stuff from the charm name (See the `rename.sh` file used in most charms).

Reply:
it's not necessarily a bad thing, it will give slow, yet reliable, building, because the first clean makes sure to remove any cruft that may be around, while the second clean avoids leaving containers around lingering.

Reply:
This is the correct pattern; as @freyes says, the initial clean is to remove any previous build artefacts (including a stopped build container) so that there is a clean build.  The second clean is to clean up the build so that resources are released; in a multi-charm build scenario the build machine may run out of disk space if this is not done.  As @samuelengineer also notes, this is also part of the official templates.